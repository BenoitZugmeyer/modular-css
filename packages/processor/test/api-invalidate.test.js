"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".invalidate()", () => {
        let processor;

        const status = (source) =>
            Object.entries(source).map(([ key, value ]) =>
                ([ relative([ key ])[0], value.valid ])
            );

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });


        it("should invalidate a relative file", async () => {
            await processor.string(
                "./simple.css",
                ".wooga { }"
            );

            processor.invalidate("./simple.css");

            expect(status(processor.files)).toMatchSnapshot();
        });

        it("should invalidate an absolute file", async () => {
            await processor.string(
                "./packages/processor/test/specimens/simple.css",
                ".wooga { }"
            );

            processor.invalidate(require.resolve("./specimens/simple.css"));

            expect(status(processor.files)).toMatchSnapshot();
        });

        it("should throw if no file is passed", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            expect(() => processor.invalidate()).toThrowErrorMatchingSnapshot();
        });

        it("should throw if an invalid file is passed", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            expect(() => processor.invalidate("nope.css")).toThrowErrorMatchingSnapshot();
        });

        it("should invalidate all dependents as well", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            processor.invalidate("./packages/processor/test/specimens/folder/folder.css");

            expect(status(processor.files)).toMatchSnapshot();
        });

        it("should reprocess invalidated files", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            processor.invalidate("./packages/processor/test/specimens/start.css");

            await processor.file("./packages/processor/test/specimens/start.css");

            expect(status(processor.files)).toMatchSnapshot();
        });
    });
});
