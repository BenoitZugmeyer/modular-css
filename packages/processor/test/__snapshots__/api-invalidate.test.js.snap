// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .invalidate() should invalidate a relative file 1`] = `
Array [
  Array [
    "simple.css",
    false,
  ],
]
`;

exports[`/processor.js .invalidate() should invalidate all dependents as well 1`] = `
Array [
  Array [
    "packages/processor/test/specimens/start.css",
    false,
  ],
  Array [
    "packages/processor/test/specimens/local.css",
    false,
  ],
  Array [
    "packages/processor/test/specimens/folder/folder.css",
    false,
  ],
]
`;

exports[`/processor.js .invalidate() should invalidate an absolute file 1`] = `
Array [
  Array [
    "packages/processor/test/specimens/simple.css",
    false,
  ],
]
`;

exports[`/processor.js .invalidate() should reprocess invalidated files 1`] = `
Array [
  Array [
    "packages/processor/test/specimens/start.css",
    true,
  ],
  Array [
    "packages/processor/test/specimens/local.css",
    true,
  ],
  Array [
    "packages/processor/test/specimens/folder/folder.css",
    true,
  ],
]
`;

exports[`/processor.js .invalidate() should throw if an invalid file is passed 1`] = `"Unknown file: nope.css"`;

exports[`/processor.js .invalidate() should throw if no file is passed 1`] = `"invalidate() requires a file argument"`;
